Week01/Problem03/Abhiraj_3.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/local/bin/python3.6
+a = 600851475143
+prime = [];
+prime.append(2)
+def nextprime():
+	n = prime[-1]
+	m = n
+	while(m == prime[-1]):
+		flag = 1
+		n = n+1
+		for i in prime:
+			if (n%i == 0):
+				flag = 0
+				break
+		if (flag == 1):
+			m = n
+	prime.append(m)
+	return prime[-1]
+
+b = a
+while(b > 1):
+	if(b%prime[-1] == 0):
+		b = b/prime[-1]
+	else:
+		nextprime()
+print("The largest prime factor of",a,"=",prime[-1])
```

Comment:
I think you could use string formatting to make your life easier here.  Python supports two "flavors" of string formatting, both [C-style](https://www.learnpython.org/en/String_Formatting) and specifically [Python 3 style](http://www.python-course.eu/python3_formatted_output.php); both styles are compared [here](https://pyformat.info/).  For example, this `print()` statement could be written like:
```
# C-Style (Printf) string formatting
print("The largest prime factor of %d = %d" % (a, prime[-1]))
# Pythonic (specifically Python 3) formatting
print("The largest prime factor of {0:d} = {1:d}".format(a, prime[-1]))
```